core/modules/qana/QservRestrictorPlugin.cc

```diff
@@ -593,6 +596,14 @@ QservRestrictorPlugin::_newSecIndexRestrictor(
     query::QsRestrictor::Ptr p;
     std::shared_ptr<query::ColumnRef> secIndex =
         resolveAsColumnRef(context, cp->left);
+    if (secIndex)  {
+        std::string theDirTable =
+            context.cssFacade->getDirTable(secIndex->db, secIndex->table);
+        if ( ! theDirTable.empty() ) {
+            LOGF_DEBUG("Setting dirTable for sIndex to %1%" % theDirTable);
+            secIndex->table = theDirTable;
+        }
+    }
```

Comment:
Sorry, I said that in a confusing way. What I meant was that you should set the db for the `secIndex` variable (a column ref) to the db of the director table. However, we might not be storing the database of a table's director table in the CSS, so you probably cannot do that. I think we really do need to support for this: an L3 child table might very well have an LSST data release Object table as its director, while almost certainly not living in the DR database. So I think this is an oversight in the CSS metadata and needs a ticket.